ui/src/app/utils/stores.tsx

```diff
@@ -259,6 +259,7 @@ export function withParamsKey(...paramNames: (keyof MatchParams)[]) {
 export interface UserAppsStore {
   updating?: boolean;
   userApps?: ListAppsResponse;
+  timeoutID?: any;
```

Comment:
The purpose of the id is so that you can keep a reference to the timer. This is important, because regardless of where polling is started, you will need to say which timer to stop when the workspace wrapper unmounts. If not, polling will continue when you navigate to other pages, and this will put your page into an unexpected state.

It is interesting about the return type of setTimeOut. [This page provides some good context. ](https://stackoverflow.com/a/51040768)[Javascript's setTimeout returns a number (a timer id)](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#return_value),but [Node's setTImeOut returns a Timeout object.](https://nodejs.org/docs/latest/api/timers.html#settimeoutcallback-delay-args) It seems as though we are using Node's version, but we run into problem's if we use Timeout as the type for the interface:
![image](https://user-images.githubusercontent.com/24514630/231273568-e307cb86-8f7a-46f2-a279-2989fa2fa77c.png)

That leaves us with a few choices:
1) Use window.setTimeout (returns number based on Javascript standard)
2) [Convert the setTimeout return to a number.](https://nodejs.org/docs/latest/api/timers.html#timeoutsymboltoprimitive)
3) [Use ReturnType ](https://stackoverflow.com/a/51040768)
4) Use any

I am open to other solutions, but this is what I have found so far.

Reply:
Thanks!  I had forgotten that the `return` from `useEffect()` was the cleanup action, so I didn't realize that was what was happening.

I also do not have a strong opinion on which option we choose.

Reply:
Thanks. I had forgotten too! I remembered componentDidUnmount from class based components but had to look up the equivalent for functional components.